### ISSUE TITLE
implicit_assignment_linter should skip assignment in while() condition under allow_lazy=TRUE

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2263]
```r
lint("
while (!is.null(row <- GetRow(state))) {
  Append(row)
  UpdateState(state)
}
", implicit_assignment_linter(allow_lazy = TRUE))
# <text>:2:17: warning: [implicit_assignment_linter] Avoid implicit assignments in function calls. For example, instead of `if (x <- 1L) { ... }`, write `x <- 1L; if (x) { ... }`.
# while (!is.null(row <- GetRow(state))) {
#                 ^~~~~~~~~~~~~~~~~~~~
```

**Comment by MichaelChirico** [comment]
Oops, I forgot the names of my own arguments :)

I meant to use `allow_scoped=TRUE`, not `allow_lazy=TRUE` 🙃
